content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: 'Prisma Client extensions'
+metaTitle: 'Prisma Client extensions (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+Use Prisma Client extensions to add functionality to your models, result objects, queries, or to Prisma Client itself.
+
+There are four types of extension. You can use each type to customize a different aspect of Prisma Client.
+
+- `model` extensions: [add custom methods or fields to your models](/concepts/components/prisma-client/client-extensions/model)
+- `client` extensions: [add client-level methods to Prisma Client](/concepts/components/prisma-client/client-extensions/client)
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To use a Prisma Client extension, you must create an associated _extended client_. An extended client is a forked variant of the standard Prisma Client with an extension applied. The standard client does not change.
+
```

Comment:
I would really like to see a basic code snippet here already. The terminology of an "extended client" and "forked variant" are quite abstract and I don't think people will have an intuition on what that means. Adding a basic code snippet that illustrates what is meant by these terms would be really helpful here.

Reply:
We do have a code snippet a bit later. The problem is that if we break this intro up with code snippets, then the other general facts about extensions and feel like an afterthought, and are more easily missed.  Really, it's meant to be a short intro to what extensions are, then we expand on that below. However, I'll look at the intro again with your point in mind, and see what I can do.

Reply:
@nikolasburk I take your point that there is a lot of conceptual material up-front.  However, for a reference doc, I think it's not unreasonable to have a conceptual introduction before we get into the "how to" material. This isn't trying to be a tutorial, so we don't want it to be _too_ narrative. With the in-page TOC, I think it's easy enough for people to skip ahead if they want to just start trying it.

Also, we're going to rethink the terminology, which might make this all a bit easier to grasp. IMO there's a problem with "extended client" - we should use a term for the forked variant that doesn't have some variation of the word "extend" in it.